presto-main/src/main/java/com/facebook/presto/connector/system/SystemSplitManager.java

```diff
@@ -60,7 +60,9 @@ public ConnectorSplitSource getSplits(ConnectorTransactionHandle transactionHand
         SystemTableHandle tableHandle = layoutHandle.getTable();
 
         TupleDomain<ColumnHandle> constraint = layoutHandle.getConstraint();
-        SystemTable systemTable = tables.get(tableHandle.getSchemaTableName());
+        SystemTable systemTable = tables.getSystemTable(session, tableHandle.getSchemaTableName())
+                // table might disappear in the meantime
+                .orElseThrow(() -> new PrestoException(NOT_FOUND, format("Table %s not found", tableHandle.getSchemaTableName())));
```

Comment:
nit. maybe you could wrap this as method in `STP`, it is repeated (two times ;))

Reply:
Lets leave it.